uptane/services/inventorydb.py

```diff
@@ -169,4 +84,122 @@ def scrub_filename(fname, expected_containing_dir):
   return abs_fname
 
 
+class InventoryDataBase():
+  """
+  Dictionary design:
+
+  {
+    'vehicle manifests':
+             {
+               'vehicle_id_1': [ <full vehicle manifests, in order of receipt> ],
+               ...
+             },
+    'ecu_manifests':
+             {
+               'ecu_serial': [ <full ECU manifests, in order of receipt>],
+               ...
+             }
+    'public_keys':
+             {
+               'vehicle_primaries':
+               {
+                 'vehicle_id_1': { ecu_id: 'primary_ecu_id',
+                                   public_key: <publickey for primary ecu>
+                                 },
+                 ...
+               },
+               'all_ecus':
+               {
+                 'ecu_serial_2': <publickey>,
+                 ...
+               }
+             }
+  }
+  """
+  # Inventory database dictionary
+  inventory_db = {'vehicle_manifests': {},
+                  'ecu_manifests': {},
+                  'public_keys': {
+                                  'vehicle_primaries': {},
+                                  'all_ecus': {}
+                                 }
+                 }
+
+  # Register ECU
+  def register_ecu(self, isPrimary, vin, ecu_serial, public_key):
+    if isPrimary:
+      vehicle_primary_ids = self.inventory_db['public_keys']['vehicle_primaries']
+      if (vehicle_primary_ids and (vin in vehicle_primary_ids)):
+        # rewrite value
+        (self.inventory_db['public_keys']['vehicle_primaries'])[vin] = \
+           {'ecu_id': ecu_serial, 'public_key': public_key}
+      else:
+        temp_dic = {'ecu_id': ecu_serial, 'public_key': public_key}
+        (self.inventory_db['public_keys']['vehicle_primaries'])[vin] = temp_dic
+    else:
+      ecu_serial_ids = self.inventory_db['public_keys']['all_ecus']
+      if (ecu_serial_ids and (ecu_serial in ecu_serial_ids)):
+        (self.inventory_db['public_keys']['all_ecus'])[ecu_serial] = public_key
+      else:
+        (self.inventory_db['public_keys']['all_ecus'])[ecu_serial] = public_key
+
+    is_registered = 1
+
+
+  # Save vehicle manifest
+  def save_vehicle_manifest(self, vin, signed_vehicle_manifest):
```

Comment:
- I mentioned that saving the vehicle manifest should also save all the contained ECU manifests, so there should be code in save_vehicle_manifest that goes through each ECU manifest inside (see uptane.formats.SIGNABLE_VEHICLE_VERSION_MANIFEST_SCHEMA) and saves each ECU manifest (calling the appropriate function here). This format is referred to later in the code as well.